### ISSUE TITLE
Fact as requirements

### ISSUE THREAD

**Issue by carduz** [sweng2/sweng2#22]
we have to write alloy facts as natural language requirements

**Comment by carduz** [comment]
@tux-mind @gdejaege

**Comment by tux-mind** [comment]
since I'm out of the index.md editing let me collect them for you.

## Facts

### codeAreUnique

taxi codes are unique ( PrimaryKey )

### allTaxiAreOwned

each taxi has been assigned to a driver

be aware of this one, I think that the company can have some taxis in magazine, to use them if a taxi get broken.

I wrote this fact just to not have Taxi instances around the alloy graph

### taxiHaveOnlyOneDriver

a.k.a one-to-one relationship

### phoneNumberAreUnique

client phone number are unique ( PrimaryKey )

### pathPositionsAreUnique

`StopPosition` that compose a `Path` must be unique ( `Path instanceof SortedSet` )

### pathSameNumberOfLoadAndDrop

a path is composed of a number of stop positions.
half of them are loading positions, the other are dropping positions.

### pathClientHasALoadAndADrop

a client referenced by a path has an equal number of load and drop positions.
( not only one, take in account a merged request with 2 request from the same client )

TODO: fix this fact

### pathStartWithLoad

a path starts with a LoadPosition

### pathEndWithDrop

a path ends with a DropPosition

### availableDriversMustBeEnqueued

all drivers marked as available must be in a queue and all drivers in the queue must be available

### enqueuedElementsMustBeUnique

elemnts of a `Queue` must be unique.

### oneQueueForZone

Each zone has it's own Driver and Request queue

### zoneOwnAllDriverQueues

All `DriverQueue` must be part of a `Zone`

### zoneOwnAllRequestQueues

All `RequestQueue` must be part of a `Zone`

### ownAllDrivers

All `Driver` must be referenced by `TaxiCentral.driver`.

in other words our system must known all drivers.

### ownAllClients

as above, for clients.

### ownAllZones

as above, for zones.

## Short Facts

### Taxi

code must be a positive number
seats can vary between 1 and 4

### Path

it contains at least 2 positions

### Request

passengers must be a positive number

### Zone

it has more than 2 `Position`s that compose it's bounds.
bounds must be composed by `Position`s and not any of it's subclasses.
bounds must be a set ( it must not contains duplicates )

**Comment by carduz** [comment]
@gdejaege I think that you should insert these as requirements

**Comment by gdejaege** [comment]
Mmh some are more domain properties.

**Comment by carduz** [comment]
right, who write? Me or you?

**Comment by gdejaege** [comment]
I can add them :) but I think the majority is already there ;)

**Comment by carduz** [comment]
you can add requirements while I add domains ;)

**Comment by carduz** [comment]
I have written short facts and phoneNumberAreUnique in domain/glossary. I think that other facts are requirements

**Comment by carduz** [comment]
done? @gdejaege

**Comment by gdejaege** [comment]
Yes it is done ;)
